### PR TITLE
200 fix hardcoding pieces in helm chart

### PR DESCRIPTION
**Pull request by longshuicy**
Load the keycloak realm in as ConfigMap. You can test by deleting the Realm in keycloak and then restart the keycloak pods see if it gets loaded correctly.

### CODE REVIEW

**Review by robkooper — CHANGES_REQUESTED** [review]
This will hard-code the realm to be for a specific instance, and will overwrite the changes that the user makes on a restart of keycloak.

File: deployments/kubernetes/charts/clowder2/files/realm.json
Comment (robkooper) [inline deployments/kubernetes/charts/clowder2/files/realm.json:1]: This is a massive file, and not sure all the parameters are needed if they are the defaults.

Comment (longshuicy) [reply deployments/kubernetes/charts/clowder2/files/realm.json:1]: there is a mini version of that but i haven't test it

**Comment by longshuicy** [comment]
> This will hard-code the realm to be for a specific instance, and will overwrite the changes that the user makes on a restart of keycloak.

Ah right... what's a better way then, using initContainers?

**Comment by robkooper** [comment]
I would have it as an optional hook (post-install) in the helm chart when you install it (not upgrade). I would have a boolean flag that is "install realm" and have it be false. This way the user can decide that you want to import it.

Problem with init is that if you are not careful it will overwrite any changes the user has made.

**Comment by longshuicy** [comment]
Created a new task for it #689 for the future.

**Review by lmarini — APPROVED** [review]

**Pull request merged**